src/opera/api/controllers/security_controller.py

```diff
@@ -0,0 +1,177 @@
+import functools
+import re
+from base64 import b64encode
+
+import connexion
+import requests
+
+from opera.api.cli import CSAR_db, SQL_database
+from opera.api.openapi.models.just_message import JustMessage
+from opera.api.settings import Settings
+
+# use connection pool for OAuth tokeninfo
+adapter = requests.adapters.HTTPAdapter(pool_connections=100, pool_maxsize=100)
+session = requests.Session()
+for protocol in Settings.introspection_protocols:
+    session.mount(protocol, adapter)
+
+role_regex = r"(?P<domain>\w+)_(?P<type>\w+)_(?P<permissions>\w)"
+AADM_SUFFIX = "aadm"
+RM_SUFFIX = "rm"
+
+
+def check_api_key(apikey, required_scopes=None):
```

Comment:
What is the point of this function? It appears to be unused. Also, why does it need required_scopes arg?

Reply:
It is mentioned in yaml spec `x-apikeyInfoFunc: opera.api.controllers.security_controller.check_api_key`
Used for checking api key.